src/pkgcheck/checks/python.py

```diff
@@ -422,3 +426,196 @@ def feed(self, pkg):
+    ``[${PYTHON_USEDEP}]`` suffix.
+
+    All argument  to ``python_has_version`` or ``has_version`` inside
+    ``python_check_deps`` should contain ``[${PYTHON_USEDEP}]`` suffix for the
+    dependency [#]_.
+
+    .. [#] https://projects.gentoo.org/python/guide/any.html#dependencies
+    """
+
+    @property
+    def desc(self):
+        return f'line: {self.lineno}: missing [${{PYTHON_USEDEP}}] suffix for argument "{self.line}"'
+
+
+class PythonMismatchUseCheckVersionAnyDep(results.VersionResult, results.Warning):
+    """Package has mismatch in dependency's use flags between call to
+    ``python_gen_any_dep`` and ``python_has_version``.
+
+    For every dependency used under ``python_gen_any_dep``, the check for a
+    matching python implementation in ``python_has_version`` should match the
+    exact use flags.
+
+    .. [#] https://projects.gentoo.org/python/guide/any.html#dependencies
+    """
+
+    def __init__(self, dep_category, dep_atom, use_flags, location, **kwargs):
+        super().__init__(**kwargs)
+        self.dep_category = dep_category
+        self.dep_atom = dep_atom
+        self.use_flags = tuple(use_flags)
+        self.location = location
+
+    @property
+    def desc(self):
+        s = pluralism(self.use_flags)
+        use_flags = ', '.join(map(str, self.use_flags))
+        return f'{self.dep_category}: mismatch for {self.dep_atom} check use flag{s} [{use_flags}] in {self.location}'
+
+
+class PythonMismatchDepCheckVersionAnyDep(results.VersionResult, results.Warning):
+    """Package has mismatch in dependencies between call to
+    ``python_gen_any_dep`` and ``python_has_version``.
+
+    For every dependency used under ``python_gen_any_dep``, a matching check
+    for a matching python implementation in ``python_has_version`` should
+    exist.
+
+    .. [#] https://projects.gentoo.org/python/guide/any.html#dependencies
+    """
+
+    def __init__(self, dep_category, dep_atom, use_flags, location, **kwargs):
+        super().__init__(**kwargs)
+        self.dep_category = dep_category
+        self.dep_atom = dep_atom
+        self.use_flags = tuple(use_flags)
+        self.location = location
+
+    @property
+    def desc(self):
+        use_flags = ', '.join(map(str, self.use_flags))
+        return f'{self.dep_category}: missing check for {self.dep_atom}[{use_flags}] in {self.location}'
+
+
+class PythonWrongUsageCheck(Check):
+    """Check ebuilds for wrong usage of python eclasses features.
+
+    This check parses the ebuild and check for more advanced failures."""
+
+    _source = sources.EbuildParseRepoSource
+    known_results = frozenset({
+        PythonHasVersionUsage,
+        PythonHasVersionMissingPythonUsedep,
+        PythonMismatchUseCheckVersionAnyDep,
+        PythonMismatchDepCheckVersionAnyDep,
+    })
+
+    has_version_known_flags = {
+        '-b': 'BDEPEND',
+        '-r': 'RDEPEND',
+        '-d': 'DEPEND',
+        '--host-root': 'BDEPEND',
```

Comment:
--host-root is only in EAPI 5 and 6 and it should map to DEPEND for those EAPIs.

Reply:
When setting in the code, if I'm in `BDEPEND` type EAPI doesn't support it, fallback to `DEPEND`